gnovm/pkg/gnolang/values_string.go

```diff
@@ -62,16 +123,42 @@ func (sv *SliceValue) String() string {
 }
 
 func (pv PointerValue) String() string {
-	// NOTE: cannot do below, due to recursion problems.
-	// TODO: create a different String2(...) function.
-	// return fmt.Sprintf("&%s", v.TypedValue.String())
-	return fmt.Sprintf("&%p.(*%s)", pv.TV, pv.TV.T.String())
+	return pv.ProtectedString(newSeenValues())
+}
+
+func (pv PointerValue) ProtectedString(seen *seenValues) string {
+	if seen.Contains(pv) {
+		return fmt.Sprintf("%p", &pv)
+	}
+
+	seen.Put(pv)
+	defer seen.Pop()
+
+	// This method was limited and not working correctly previously. Allowing for it to work as
+	// intended means that it needs to ensure the type value is not nil before attempting to
+	// convert it to a string.
```

Comment:
nit: interesting for a future Gno historian, but not so much for a code reader?

```suggestion
	// Handle nil TV's, avoiding a nil pointer deref below.
```